pkg/bgpv1/cell.go

```diff
@@ -54,3 +56,24 @@ func newBGPPeeringPolicyResource(lc hive.Lifecycle, c client.Clientset, dc *opti
 			c.CiliumV2alpha1().CiliumBGPPeeringPolicies(),
 		))
 }
+
+func newSecretResource(lc hive.Lifecycle, c client.Clientset, dc *option.DaemonConfig) resource.Resource[*slim_core_v1.Secret] {
+	// Do not create this resource if the BGP Control Plane is disabled
+	if !dc.BGPControlPlaneEnabled() {
+		return nil
+	}
+
+	if !c.IsEnabled() {
+		return nil
+	}
+
+	// Do not create this resource if the BGP namespace is not set
+	if dc.BGPSecretsNamespace == "" {
+		return nil
```

Comment:
Yes, sorry, I hadn't considered how this would be configured, but now I've made the helm chart changes it clear we'd expect this to be configured in most cases, so a log does make sense.